samples/MQTTSN/src/paho_iot_hub_telemetry_example.c

```diff
@@ -0,0 +1,669 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#ifdef _WIN32
+#include <Windows.h>
+#else
+#include <unistd.h>
+#endif
+
+#include "MQTTSNPacket.h"
+#include "azure/iot/az_iot_hub_client.h"
+#include "transport.h"
+
+// DO NOT MODIFY: Device ID Environment Variable Name
+#define ENV_DEVICE_ID "AZ_IOT_DEVICE_ID"
+
+// DO NOT MODIFY: IoT Hub Hostname Environment Variable Name
+#define ENV_IOT_HUB_HOSTNAME "AZ_IOT_HUB_HOSTNAME"
+
+// DO NOT MODIFY: MQTTSN Gateway IP Address Environment Variable Name
+#define ENV_MQTTSN_GATEWAY_ADDRESS "MQTTSN_GATEWAY_ADDRESS"
+
+// DO NOT MODIFY: MQTTSN Gateway Port Environment Variable Name
+#define ENV_MQTTSN_GATEWAY_PORT "MQTTSN_GATEWAY_PORT"
+
+#define DEFAULT_GATEWAY_ADDRESS "127.0.0.1"
+#define DEFAULT_GATEWAY_PORT "10000"
+#define TELEMETRY_SEND_INTERVAL 1 // seconds
+#define NUMBER_OF_MESSAGES 5
+#define TELEMETRY_PAYLOAD \
+  "{\"d\":{\"myName\":\"IoT mbed\",\"accelX\":12,\"accelY\":4,\"accelZ\":12,\"temp\":18}}"
+
+#ifdef AZ_TELEMETRY_QOS_0
+#undef ENABLE_PUBACK // default to qos 1 and enable puback if QoS 1
+#else
+#define ENABLE_PUBACK
+#endif
+
+static char topic_name[128];
+static char device_id[64];
+static char iot_hub_hostname[128];

```

Comment:
move them to context:


so context will look like 

typedef struct iothub_client_context_tag
{
  char host[128];
  int port;
  char client_id[64];
  unsigned short telemetry_topic_id;
  az_iot_hub_client client;
  short packet_id;
} iothub_client_context;

Reply:
Since we are not using the iot_hub_hostname other than for intializing the hub client, should we just use the scratch buffer for this? instead of saving to context? (because we'll never be accessing it from context)